dxp/react_rendering.md

```diff
@@ -0,0 +1,148 @@
+2. A `Map` containing arbitrary data that will be injected to the React component as initial properties. Note that the data inside the map must be compatible with JavaScript, so you cannot inject any Java object you like, only primitive types, `Map`s, `Array`s, ... that can be serialized to JSON.
+3. The current `HttpServletRequest` for which the component is being rendered.
+4. A `Writer` where the rendered HTML (and optionally the JavaScript) is written to.
+
+Note that parameter 4 is not intended to be used to write the rendered HTML and JavaScript to any place you like but to pass the `PrintWriter` from the `HttpServletResponse` or the `JspWriter` from the `PageContext` you have available when invoking the method.
+
+Also, take into account that, if you don't render the JavaScript inline (see option `positionInline` of `ComponentDescriptor`), the JavaScript will be written to the bottom of the current `HttpServletResponse`, while the HTML will be inlined into the passed `Writer`.
+
+### What the ComponentDescriptor contains
+
+`ComponentDescriptor` instances are [value objects](https://en.wikipedia.org/wiki/Value_object) that describe what needs to be rendered. A `ComponentDescriptor` may have up to five fields (of which only the first one is mandatory):
+
+1. `module`: the name of the AMD module that exports the React component to be rendered. Note that you need to provide the canonical name of the module (including its version number). To avoid hardcoding any version number you may use the instance of the `com.liferay.frontend.js.loader.modules.extender.npm.NPMResolver` service tied to the OSGi bundle containing the AMD module to be rendered.
+2. `componentId`: the `componentId` that will be given to the component's JavaScript as an initial property. Note that this field is only used if the `data` map passed to `renderReact` does **NOT** contain a `componentId` property.
+3. `dependencies`: this is deprecated; don't use it 🧐.
+4. `positionInline`: a flag to render the JavaScript directly in the current position of the HTML response or defer it to the bottom of the page. By default `false`.
+5. `propsTransformer`: optional name of an AMD module exporting a function to transform the properties before they are passed to the component.
+
+### How the React component is rendered
+
+Say you run the following code:
+
+```java
+ComponentDescriptor componentDescriptor =
+	new ComponentDescriptor(
+		"my-bundle@1.0.0/component/button", "myButton", null, false,
+		"my-bundle@1.0.0/propsTransformer");
+
+Map<String, Object> data = new HashMap<>();
+
+data.put("color", "red");
+
+reactRenderer.renderReact(
+	componentDescriptor, data, httpServletRequest,
+	httpServletResponse.getWriter());
+```
+
+This will cause HTML like this to be rendered in the current position of the `HttpServletResponse`:
+
+```html
+<div id="XJAZ"></div>
+```
+
+(where `XJAZ` is a random placeholder id).
```

Comment:
I'm missing a bit of an explanation about this placeholder element. We use it to mount the react component in the parent element, which is done through the `render` method. This is a tricky aspect of it that we should collect here, I think.